packages/@react-aria/autocomplete/docs/useAutocomplete.mdx

```diff
@@ -59,6 +62,132 @@ to filter the collection. `useAutocomplete` handles exposing the correct ARIA at
 State is managed by the <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useAutocompleteState} /> hook from `@react-stately/autocomplete`.
 The state object should be passed as an option to `useAutocomplete`.
 
+
+## Example
+
+This example uses [React Aria Components](../react-aria/components.html) and shows how, if you need to build your own `Autocomplete` component, you can integrate it with existing React Aria Components.
+This `Autocomplete` uses a [SearchField](./SearchField.html) as the input and controls a [ListBox](./ListBox.html).
+See those docs for examples of what they can do or how to customize them. This `Autocomplete` renders no dom, so see those components for styling as well.
+
+```tsx example
+import {AriaAutocompleteProps, useAutocomplete} from '@react-aria/autocomplete';
+import {useAutocompleteState} from '@react-stately/autocomplete';
+import {useRef} from 'react';
+import {
+  Label,
+  Text,
+  InputContext,
+  Input,
+  Provider,
+  SearchFieldContext,
+  SearchField,
+  ListBox,
+  ListBoxItem,
+  useFilter,
+  UNSTABLE_InternalAutocompleteContext
```

Comment:
Seems like the main thing to determine is whether we want to export this or if we want to change approaches. For example, we could consider providing ListBoxContext and MenuContext instead of having those components consume InternalAutocompleteContext. Not sure which is better. The same kind of thing applies to TextFieldContext and SearchFieldContext and we chose the opposite approach there.

Reply:
Yeah, that's a good question. I like the idea that users aren't restricted to our components (without basically implementing what is in this page to add/remove other contexts).
I'm not sure why we went the other way on the inputs, do you or @LFDanLu have a short summary of that? Or did it just kind of happen?

Reply:
I don't quite remember the exact reasoning, but I recall there being a concern with how many collection components may end up being supported by Autocomplete (table, gridlist, listbox, etc) and thus the number of individual contexts that would be needed vs a single one. I'd be happy to shift to individual contexts if only to reuse the existing collection contexts + match our TextField/SearchField context usage here

Reply:
Technically we have the same issue with TextField/SearchField, it's possible there may be other ways to control the filter (maybe a predefined set of radio options/tags?)
So how do we want to handle that as well?

Reply:
it feels like input fields are smaller in number (radios/tags would take a separate set of props than the ones coming via textFieldProps from useAutocomplete) but we'd still use individual contexts there (aka radioGroupContext/tagGroupContext or something)